### ISSUE TITLE
replacing file

### ISSUE THREAD

**Issue by rounk-ctrl** [echnobas/sxsc#5]
is it possible, to make it "update" a file with a custom file?

**Comment by echnobas** [comment]
> is it possible, to make it "update" a file with a custom file?
> 
> 
> 
> i tried something like adding the file to the cdf file and modifying the manifest, but 
> 
> 
> 
> ![image](https://github.com/echnobas/sxsc/assets/70931017/5d00c7aa-0636-4c1d-9cc3-63f5a17a15d3)
> 
> 

yes i have done this. the manifest expects a base64 encoded sha256 hash of the file and the file must be included in the catalog file.

**Comment by echnobas** [comment]
i should mention. replacing a file owned by another package is undefined behaviour because microsoft obviously doesn't intend for other people to be making cbs packages.

**Comment by rounk-ctrl** [comment]
can you send the manifest?

Also does it auto create the hardlink to the respective folder after replacing it?

**Comment by rounk-ctrl** [comment]
also is this how you include the file in the catalog? because it seems it isnt getting added there
```
<HASH>F4=amd64_microsoft-windows-accessibilitycpl_31bf3856ad364e35_69.69.0.0_none_83ae67a5cf491e0f\accessibilitycpl.dll.mun

```

**Comment by echnobas** [comment]
> can you send the manifest?
> 
> 
> 
> Also does it auto create the hardlink to the respective folder after replacing it? like say if i replace accessibilitycpl.dll.mun in winsxs, will the file in SystemResources also get updated?

i don't have a manifest i can give you but if you open any update as a cab you'll figure it out

it'll make the hard link but sfc will be very confused because two packages "owning" a file is an invariant cbs expects microsoft to uphold. what ends up happening from my experience is they fight over it and the copy of the file from one package will overwrite the link so i don't recommend this.

**Comment by echnobas** [comment]
> also is this how you include the file in the catalog? because it seems it isnt getting added there
> 
> ```
> 
> <HASH>F4=amd64_microsoft-windows-accessibilitycpl_31bf3856ad364e35_69.69.0.0_none_83ae67a5cf491e0f\accessibilitycpl.dll.mun
> 
> 
> 
> ```

this is correct, i don't think the name after the hash matters. but if it does then F(N) must have appeared to me in a dream because i have no idea where i pulled that from.

**Comment by rounk-ctrl** [comment]
> i don't have a manifest i can give you but if you open any update as a cab you'll figure it out
> 
> it'll make the hard link but sfc will be very confused because two packages "owning" a file is violating an invariant cbs expects microsoft to uphold. what ends up happening from my experience is they fight over it and the copy of the file from one package will overwrite the link so i don't recommend this.

```
<file name="accessibilitycpl.dll.mun" destinationPath="$(runtime.windows)\SystemResources\"
		sourceName="accessibilitycpl.dll.mun" importPath="$(build.nttree)\" sourcePath=".\">
		<securityDescriptor name="WRP_FILE_DEFAULT_SDDL" />
		<asmv2:hash xmlns:asmv2="urn:schemas-microsoft-com:asm.v2">
			<dsig:Transforms xmlns:dsig="http://www.w3.org/2000/09/xmldsig#">
				<dsig:Transform Algorithm="urn:schemas-microsoft-com:HashTransforms.Identity" />
			</dsig:Transforms>
			<dsig:DigestMethod xmlns:dsig="http://www.w3.org/2000/09/xmldsig#"
				Algorithm="http://www.w3.org/2000/09/xmldsig#sha256" />
			<dsig:DigestValue xmlns:dsig="http://www.w3.org/2000/09/xmldsig#">
			YTkyOTAwMzk4ZDZlZTE2YjVhN2M2Y2E3ODU2NjdjYTdhYTQxZDZhZDk0ODczOThkODJjN2Y5MzJkY2ZkMWNhNg==</dsig:DigestValue>
		</asmv2:hash>
	</file>
```

doesnt seem to work

**Comment by echnobas** [comment]
after running sxsc.py, edit the manifest to your liking, add the file to files.txt and update.cdf, then run build.bat as normal

**Comment by rounk-ctrl** [comment]
thats what i did

**Comment by rounk-ctrl** [comment]
![image](https://github.com/echnobas/sxsc/assets/70931017/b17ae752-e493-431d-b49d-1faeed16b54c)
oo i got somewhere

**Comment by rounk-ctrl** [comment]
now what
